lib/report_plugins/autotune.js

```diff
@@ -0,0 +1,333 @@
+    $('<td>' + recommended + '</td>').appendTo(tr);
+  }
+  var td = $('<td/>')
+  appendArrowIcon(td, current, recommended);
+  td.appendTo(tr);
+  tr.appendTo(table);
+}
+
+function getCurrentISF(store) {
+  return store.sens[0].value;
+}
+
+function getCurrentCSF(store) {
+  return getCurrentISF(store) / getCurrentCarbRatio(store);
+}
+
+function getCurrentCarbRatio(store) {
+  return store.carbratio[0].value;
+}
+
+function roundNum(num) {
+  return Number(num).toFixed(3);
+}
+
+function appendArrowIcon(td, current, recommended) {
+  var icon = $('<img/>');
+  if (recommended > current) {
+    $('<img src="/images/up_arrow.png" />').appendTo(td);
+  } else if (recommended < current) {
+    $('<img src="/images/down_arrow.png" />').appendTo(td);
+  }
+}
+
+function findActiveProfiles(profiles, sorteddaystoshow) {
+  var result = {};
+  result.activeIndex = -1;
+  result.manyActive = false;
+
+  var dateFrom = new Date(sorteddaystoshow[0]);
+  var dateTo = new Date(sorteddaystoshow[sorteddaystoshow.length-1]);
+  var sortedProfiles = [];
+  for (var i = 0; i < profiles.length; i++) {
+    var record = {};
+    record.profileIndex = i;
+    if (! profiles[i].startDate) { continue; }
+    var splittedString = profiles[i].startDate.split("T");
+    record.startDate = new Date(splittedString[0]);
+    sortedProfiles.push(record);
+  }
+  sortedProfiles.sort(function (r1, r2) {
+    return (r1.startDate < r2.startDate) ? -1 : ((r1.startDate > r2.startDate) ? 1 : 0);
+  });
+  for (i = 0; i < sortedProfiles.length; i++) {
+    var profile = sortedProfiles[i];
+    if (profile.startDate <= dateFrom) {
+      result.activeIndex = profile.profileIndex;
+      continue;
+    }
+    if (profile.startDate >= dateFrom && profile.startDate <= dateTo)
+      result.manyActive = true;
+    if (profile.startDate > dateFrom)
+      break;
+  }
+
+  return result;
+}
+
+function getHourMinuteText(hour, minute)
+{
+  var text = "";
+  if (hour < 10)
+    text += "0";
+  text += hour.toString();
+  text += ":";
+  if (minute < 10)
+    text += "0";
+  text += minute.toString();
+  return text;
+}
+
+function findBasalRecord(data, time)
+{
+  for(var i = 0; i < data.length; i++)
+  {
+    if (data[i].time == time)
+      return data[i];
+  }
+  return null;
+}
+
+function findRecommendedRecord(data, minutes) {
+  for(var i = 0; i < data.length; i++)
+  {
+    if (data[i].minutes == minutes)
+      return data[i];
+  }
+  return null;
+}
+
+autotune.report = function report_autotune(datastorage, sorteddaystoshow, options) {
+  var Nightscout = window.Nightscout;
+  var client = Nightscout.client;
+  var translate = client.translate;
+  var report_plugins = Nightscout.report_plugins;
+
+  if (!$('#compute_autotune').is(':checked')) return;
+
+  var profileRecords = datastorage.profiles;
+
+  var report = $('#autotune-report');
+  report.empty();
+
+  var foundActiveProfile = findActiveProfiles(datastorage.profiles, sorteddaystoshow);
```

Comment:
I believe this won't play nice with how Nightscout currently manages profiles, where the profile can also be enacted by a Profile Switch careportal event, so the only way to reliably get the profile that was in use is by loading the Nightscout sandbox & querying it's built-in profile manager as to what profile setting was in use at the time. Some users can also have the circadian profile enabled, which will probably confuse this implementation. @MilosKozak I recall there's no simple way to detect if circadian profile is in use, is there? We might want to just not calculate autotune at all to users who might have a profile on that's not tested with this logic.